compiler/compiler.go

```diff
@@ -716,6 +716,10 @@ func (c *compilerContext) createPackage(irbuilder llvm.Builder, pkg *ssa.Package
 			// Create the function definition.
 			b := newBuilder(c, irbuilder, member)
 			if member.Blocks == nil {
+				if cd, ok := compDefs[b.llvmFn.Name()]; ok {
+					// This is a compiler-defined function.
+					b.fillCompDef(cd)
+				}
```

Comment:
We have lots of compiler-overridden functions already. We currently replace `*ssa.Call` instructions to them with their value inline, but perhaps it makes sense to replace the contents of the functions instead (at least, in some cases). See the `syscall.Syscall*` functions. In any case, I think we should be consistent and do one or the other, I would be fine with changing most of those call replacements to replacing the definition of functions as a refactor.

Reply:
I think I can turn this into a replacement in this case.